sympy/integrals/intpoly.py

```diff
@@ -0,0 +1,806 @@
+        elif expr.is_Pow:
+            if separate:
+                monoms.append(expr)
+            else:
+                poly_dict[expr.args[1]] = expr
+        else:
+            if separate:
+                monoms.append(expr)
+            else:
+                poly_dict[1] = expr
+    else:
+        if separate:
+            monoms.append(expr)
+        else:
+            poly_dict[0] = expr
+
+    if separate:
+        return monoms
+    return poly_dict
+
+
+def clockwise_sort(poly):
+    """Returns the same polygon with points sorted in clockwise order.
+
+    Note that it's necessary for input points to be sorted in some order
+    (clockwise or anti-clockwise) for the algorithm to work. As a convention
+    algorithm has been implemented keeping clockwise orientation in mind.
+
+    Parameters
+    ==========
+    poly: 2-Polytope
+
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import clockwise_sort
+    >>> from sympy.geometry.point import Point
+    >>> from sympy.geometry.polygon import Polygon
+    >>> clockwise_sort(Polygon(Point(0, 0), Point(1, 0), Point(1, 1)))
+    Triangle(Point2D(1, 1), Point2D(1, 0), Point2D(0, 0))
+
+    """
+    n = len(poly.vertices)
+    vertices = list(poly.vertices)
+    center = Point(sum(map(lambda vertex: vertex.x, poly.vertices)) / n,
+                   sum(map(lambda vertex: vertex.y, poly.vertices)) / n)
+
+    def compareTo(a, b):
+        if a.x - center.x >= 0 and b.x - center.x < 0:
+            return -1
+        elif a.x - center.x < 0 and b.x - center.x >= 0:
+            return 1
+        elif a.x - center.x == 0 and b.x - center.x == 0:
+            if a.y - center.y >= 0 or b.y - center.y >= 0:
+                return -1 if a.y > b.y else 1
+            return -1 if b.y > a.y else 1
+
+        det = (a.x - center.x) * (b.y - center.y) -\
+              (b.x - center.x) * (a.y - center.y)
+        if det < 0:
+            return -1
+        elif det > 0:
+            return 1
+
+        first = (a.x - center.x) * (a.x - center.x) +\
+                (a.y - center.y) * (a.y - center.y)
+        second = (b.x - center.x) * (b.x - center.x) +\
+                 (b.y - center.y) * (b.y - center.y)
+        return -1 if first > second else 1
+
+    return Polygon(*sorted(vertices, key=cmp_to_key(compareTo)))
+
+
+def norm(point):
+    """Returns the Euclidean norm of a point from origin.
+
+    Parameters
+    ==========
+    point: This denotes a point in the dimensional space.
+
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import norm
+    >>> from sympy.geometry.point import Point
+    >>> norm(Point(2, 7))
+    sqrt(53)
+
+    """
+    h = 0
+    half = S(1)/2
+    if isinstance(point, tuple):
+        h = (point[0] ** 2 + point[1] ** 2) ** half
+    elif isinstance(point, Point):
+        h = (point.x ** 2 + point.y ** 2) ** half
+    elif isinstance(point, dict):
+        s = 0
+        for i in point.values():
```

Comment:
We can instead use `s = sum([i**2 for i in point.values()])`, it will also take care of empty `dict` (not need to worry about that).